File: leetcode-58-Length_of_Last_Word/leetcode-58-Length_of_Last_Word.cpp
#include <iostream>
#include <cstdlib>
#include <string>

int lengthOfLastWord(std::string s) {
    int i = s.size()-1;
    while(0<=i && s[i] == ' ')
    {
        --i;
    }
    
    int res = 0;
    while(0<=i && s[i] != ' ')
    {
        ++res;
        --i;
    }
    return res;
}


int main(){
    std::string s = "Hello World";

    auto res = lengthOfLastWord(s);
    
    std::cout << res << std::endl;
    
    return EXIT_SUCCESS;
}

File: leetcode-58-Length_of_Last_Word/leetcode-58-Length_of_Last_Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        s = s.strip()
        i: int = len(s) - 1
        while i >= 0 and s[i] != ' ': i -= 1

        return (len(s)-1) - i


if __name__ == "__main__":
    s: str = "Hello World"
    result = Solution().lengthOfLastWord(s)
    print(result)
